kyuubi-server/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala

```diff
@@ -83,18 +83,39 @@ class ExecuteStatement(
 
   private def waitStatementComplete(): Unit = try {
     setState(OperationState.RUNNING)
-    var statusResp = client.GetOperationStatus(statusReq)
+    var statusResp: TGetOperationStatusResp = null
+    var currentAttempts = 0
+
+    def getOperationStatusWithRetry: Unit = {
+      try {
+        statusResp = client.getOperationStatus(_remoteOpHandle)
+        currentAttempts = 0 // reset attempts whenever get touch with engine again
+      } catch {
+        case e: TException if currentAttempts >= maxStatusPollOnFailure =>
+          error(s"Failed to get ${session.user}'s query[$getHandle] status after" +
+            s" $maxStatusPollOnFailure times, aborting", e)
+          throw e
+        case e: TException =>
+          currentAttempts += 1
+          warn(s"Failed to get ${session.user}'s query[$getHandle] status" +
+            s" ($currentAttempts / $maxStatusPollOnFailure)", e)
+          Thread.sleep(500)
```

Comment:
Configurable?

Reply:
Look fine to me to be hardcoded. As users already have operation.status.polling.timeout